handler/implementations/maxIntBaseHandler.go

```diff
@@ -212,6 +218,11 @@ func (h *MaxIntBaseHandler) fetchFile(
 	n domain.IOnodeIface,
 	c domain.ContainerIface) (string, error) {
 
+	// We need the per-resource lock since we are about to access the resource on
+	// the host FS. See pushFile() for a full explanation.
+	h.Lock.Lock()
+	defer h.Lock.Unlock()
```

Comment:
Let's return after the if-errors instead to reduce critical section, logrus is known to be slow.

Reply:
Sure, done.